packages/react/src/components/PaginationNav/PaginationNav.js

```diff
@@ -0,0 +1,363 @@
+    back: backHidden,
+  };
+};
+
+export default class PaginationNav extends Component {
+  constructor(props) {
+    super(props);
+    const { page } = this.props;
+
+    this.state = {
+      page,
+      cuts: {
+        front: 0,
+        back: 0,
+      },
+      prevPage: page,
+    };
+
+    this.jumpToItem = this.jumpToItem.bind(this);
+    this.jumpToNext = this.jumpToNext.bind(this);
+    this.jumpToPrevious = this.jumpToPrevious.bind(this);
+  }
+
+  static propTypes = {
+    /**
+     * Additional CSS class names.
+     */
+    className: PropTypes.string,
+
+    /**
+     * The description for the backward icon.
+     */
+    backwardText: PropTypes.string,
+
+    /**
+     * The description for the forward icon.
+     */
+    forwardText: PropTypes.string,
+
+    /**
+     * The label that appears before the page number in browser tooltips and screen readers.
+     */
+    itemLabel: PropTypes.string,
+
+    /**
+     * The callback function called when the current page changes.
+     */
+    onChange: PropTypes.func,
+
+    /**
+     * The total number of items.
+     */
+    totalItems: PropTypes.number,
+
+    /**
+     * The number of items to be shown.
+     */
+    itemsShown: PropTypes.number,
+
+    /**
+     * The current page.
+     */
+    page: PropTypes.number,
+
+    /**
+     * Whether user should be able to loop through the items when reaching first / last.
+     */
+    loop: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    backwardText: 'Previous',
+    forwardText: 'Next',
+    page: 0,
+    itemLabel: 'Page',
+    itemsShown: 10,
+    loop: false,
+  };
+
+  static getDerivedStateFromProps({ page, totalItems, itemsShown }, state) {
+    const { page: currentPage, prevPage } = state;
+
+    const itemsThatFit = itemsShown >= 4 ? itemsShown : 4;
+    const pageChanged = page !== prevPage;
+    const cuts = getCuts(totalItems, itemsThatFit, currentPage);
+
+    return !pageChanged
+      ? {
+          cuts,
+        }
+      : {
+          page: (pageChanged && page) || currentPage,
+          prevPage: page,
+          cuts,
+        };
+  }
+
+  jumpToItem(index) {
+    const { totalItems } = this.props;
+
+    if (index >= 0 && index < totalItems) {
+      this.setState({
+        page: index,
+      });
+
+      this.props.onChange(index);
+    }
+  }
+
+  jumpToNext() {
+    const { totalItems, loop } = this.props;
+    const { page } = this.state;
+
+    const nextIndex = page + 1;
+
+    if (nextIndex >= totalItems) {
+      if (loop) {
+        this.jumpToItem(0);
+      }
+    } else {
+      this.jumpToItem(nextIndex);
+    }
+  }
+
+  jumpToPrevious() {
+    const { totalItems, loop } = this.props;
+    const { page } = this.state;
+
+    const previousIndex = page - 1;
+
+    if (previousIndex < 0) {
+      if (loop) {
+        this.jumpToItem(totalItems - 1);
+      }
+    } else {
+      this.jumpToItem(previousIndex);
+    }
+  }
+
+  renderDirectionButton(direction) {
```

Comment:
Would it make sense for this to be a component that exists in the `PaginationNav` file?